---
title: "Cancer Competition Analysis - Streamlined"
author: "Patrick C.N. Martin"
date: "`r Sys.Date()`"
output: 
    pdf_document:
        toc: true
        fig_crop: no
        dev: cairo_pdf
---


```{r setup, eval = TRUE, echo = FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
knitr::opts_chunk$set(dev.args = list(png = list(type = "cairo")))
```
```{r libs, eval = TRUE, echo  = FALSE, message = FALSE, warning = FALSE}

#-----------------------------------------------------------------------------#
# load base libraries
#-----------------------------------------------------------------------------#
library(ggplot2)
library(patchwork)
library(ggpubr)
library(future)
library(rjson)
library(dplyr)
library(future)
library(Matrix, lib.loc = "/common/martinp4/R")
library(ggplot2)
library(patchwork)
library(deldir)
library(imager)
library(imagerExtra)
library(hdf5r,lib.loc = "/common/martinp4/R" )
library(lpSolve, lib.loc = "/common/martinp4/R")
library(TreeDist, lib.loc = "/common/martinp4/R")
library(mclust, lib.loc = "/common/martinp4/R")
library(mcclust, lib.loc = "/common/martinp4/R")
library(pwr, lib.loc = "/common/martinp4/R")
library(gsignal, lib.loc = "/common/martinp4/R")
library(kohonen, lib.loc = "/common/martinp4/R")
library(registry, lib.loc = "/common/martinp4/R")
library(rngtools, lib.loc = "/common/martinp4/R")
library(NMF, lib.loc = "/common/martinp4/R")
library(spatstat.utils, lib.loc = "/common/martinp4/R")
library(vesalius, lib.loc = "/common/martinp4/R")
library(readxl, lib.loc = "/common/martinp4/R")
library(RColorBrewer)

# NOTE: need to use module load hdf5 first!!!!!
set.seed(42)
#-----------------------------------------------------------------------------#
# Load Analysis library
#-----------------------------------------------------------------------------#
library(Kuresi, lib.loc = "/common/martinp4/R")

#-----------------------------------------------------------------------------#
# base directories and where should plots be saved
#-----------------------------------------------------------------------------#
input <- "/common/wonklab/visium_cancer/"

output <- "/common/wonklab/visium_cancer/report/"

#-----------------------------------------------------------------------------#
# Go multicore or no?
#-----------------------------------------------------------------------------#
#plan(sequential) # single core
plan(multicore, workers = 1) # Linux and Mac
#plan(multisession, workers = 2) # Windows
max_size <- 1000 * 1024^2
options(future.globals.maxSize = max_size)
```

# Cancer Compitition in Spatial transcriptomics

Based on an extensive literature search, we defined a set of genes that were associated with cell competition/fitness. 
More specifically, these genes can be categorized into 2 categories - *win* and *lose* genes. The combination 
of high expression of *win* genes and low expression of *lose* genes are associated with a fit cell, a cell that 
will survive. While the opposite expression pattern will push the cell towards death.

```{r gene_list, eval = TRUE, echo = FALSE}
win_lose <- win_lose_genes()
win <- win_lose$win
lose <- win_lose$lose
pancreas <- cancer_maker_list(type = "pancreas")
breast <- cancer_maker_list(type = "breast")
ovary <- cancer_maker_list(type = "ovary")
```

The *win* set of genes is the following:
```{r win_g, eval = TRUE, echo = FALSE}
win
```


The *lose* set of genes is the following:
```{r lose_g, eval = TRUE, echo = FALSE}
lose
```

We sought to use this gene list to uncover competition between cancer clones (tumors) and potential find clones that are
more agressive than others within Spatial Resolved Transcriptomic (SRT) data sets.

## Spatial Transcriptomic Data
We selected four candidate data sets taken from 10X's data base of publically available Visium assays.
In particular:

* Human Ovarian Cancer 
* Human Pancreatic Cancer
* Human Breast Cancer
* Human Breast Cancer - Xenium

[10X Database](https://www.10xgenomics.com/datasets?query=&page=1&configure%5BhitsPerPage%5D=50&configure%5BmaxValuesPerFacet%5D=1000)

## Selecting Cancer using marker gene expression

From these data sets, we filtered out the stroma to only retain spots containing cancer tissue. This was achieved by using the expression 
of tissue specific cancer related markers. 


For ovary:
```{r ovary_genes, eval = TRUE, echo = FALSE}
ovary
```

For pancreas:
```{r pan_genes, eval = TRUE, echo = FALSE}
pancreas
```

For breast:
```{r breast_genes, eval = TRUE, echo = FALSE}
breast
```

As a point of comparison, we also used Seurat's module score which defines how *"active"* a set of genes is compared to a random set of genes. 
However, this approach did not yield clear results and did not match with the H&E images that clearly show the location of cancer cells. 

[Seurat Module Score](https://satijalab.org/seurat/reference/addmodulescore)


## Competition Score

### Gene Ratio

To effectively investigate cancer competition in SRT data, we develop a competition score that would highlight 
the expression ratio of *win* genes over *lose* genes. For each spot, we define its competitive potential as:

$$C = \frac{(\frac{1}{w} \sum_{i=1}^{w} W_i \cdot a_i) + 1}{ (\frac{1}{l} \sum_{j=1}^{l} L_j \cdot b_j) + 1}$$

with:

* $w$, number of win genes

* $l$, number of lose genes

* $W$, expression of win gene *i*

* $L$, expression of lose genes *j*

* $a$, gene weight associted with gene *i*
    
* $b$, gene weight associated with gene *j*

Simply, the competition score is the ratio of the average expresion of win genes over the average expression
of lose genes. We add a pseudo-count of 1 to avoid 0 division in the case that these genes are not expressed at all.
Finally, we min/max normalized the scores to highlight the least to most agressive cancer spots withith SRT data.
We also provide a ranked version of competition scores to facilitate plot reading and interpretation. 

### Gene Weights

Expert assessment of the gene sets highlighted the need to assign a weight to some of the genes since some play 
a more signficant role in cell competition. 
We manually assigned the following weights to the gene set. All genes receive a weight of 1 by default. 
```{r weights, eval = TRUE, echo = TRUE}
# Get the counts and create weight vector
weight <- rep(1, length(c(win, lose)))
names(weight) <- c(win, lose)
weight[names(weight) %in% toupper(c("Myc", "Mycn", "Ras", "Egfr"))] <- 2
weight[names(weight) == toupper("Cacfd")] <- 3
```

## Isolating Cancer clones 
To isolate competition territories within a variety of SRT cancer data sets, we used Vesalius - an R package designed to isolate
tissue territories. Specifically, we generated territories by using *win/lose* genes during the dimensionality
reduction (PCA) step of the Vesalius workflow. Next, we applied Vesalius to process the SRT data sets and isolate spatial
territories. The final territories show tissue areas characterized by unique win/lose expression patterns. 


[Vesalius R Package](https://github.com/patrickCNMartin/Vesalius)
[Vesalius Publication](https://www.embopress.org/doi/full/10.15252/msb.202211080)

## Visualization
To more effectively visualize the cancer clone territories, we assigned the average and normalized scores across all spots belonging to a
cancer territory. This approach was uses for both the gene ratio score and the gene difference score and intends to minizmise the effect of territory 
size on the final outcomes. 

When necessary, we binned the data into discrete ranges in order to better highlight score differences. We also employ score ranks for a clearer
interpretation of score and their associated territories. 

# Results & Summary

## Ovarian Cancer

```{r load_ovary, eval = TRUE, echo = FALSE}
vesalius <- load_visium("/common/wonklab/visium_cancer/data/Human_Ovarian_cancer",
    tag = "Human_Ovarian")


max_range <- get_max_range(vesalius)
vesalius <- vesalius %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 2, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.05, min_spatial_index = 5, verbose = F)

vesalius <- module_score(vesalius,
    gene_list = ovary,
    add_name = "counts")
vesalius <- compress_score(vesalius,
    trial = "counts",
    add_name = "counts_bin",
    rank = FALSE, bins = 10)

outcomes <- Kuresi::compute_competition_outcomes(vesalius, expressed = win , repressed = lose)

ov <- vesalius@territories$barcodes[vesalius@territories$counts > 10]
vesalius_f <- filter_assay(vesalius, cells = ov)

max_range <- get_max_range(vesalius_f)
vesalius_f <- vesalius_f %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 2, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.25, min_spatial_index = 5, verbose = F)



vesalius_f <- compute_ratio_score(vesalius_f,
    group_1 = win,
    group_2 = lose,
    score_type = "mean_ratio",
    add_name = "mean_ratio")

```




```{r ovary_plot, eval=TRUE, echo = FALSE}
hne <- territory_plot(vesalius, trial = "Territory",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 1, cex = 15) + 
        labs(title = "H & E - Ovarian Cancer", color = "Territories")+
        theme_void()+
        theme(
            legend.position = "left",
            plot.tag = element_text(size = 30),
            plot.margin = margin(0, 0, 0, 0, "cm"),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

cancer_counts <- score_plot(vesalius, trial = "counts_bin",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 1.4, cex = 20) + 
        labs(title = "Ovarian Cancer Marker Count", color = "Cancer Marker Counts")+
        theme_void()+
        theme(
            legend.position = "right",
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.tag = element_text(size = 30),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

pal <- colorRampPalette(rev(c("#850101","#cd8878", "#f1f1b1","#9CAAC4","#1F3B70")))
n_groups <- length(unique(vesalius_f@territories$mean_ratio))
cols <- pal(n_groups)
mean_ratio <- territory_plot(vesalius_f, trial = "mean_ratio",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 1.4, cex = 20) + 
        labs(title = "Competitivity",color = "Win/Lose Ratio")+
        scale_color_manual(values = cols)+
        theme_void()+
        theme(
            plot.tag = element_text(size = 30),
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

```



```{r ovary_plot_out, eval = TRUE, echo = FALSE, fig.width=25, fig.height=10, fig.cap = "Ovarian Cancer"}
(hne + cancer_counts + mean_ratio) +
    plot_annotation(tag_levels = "A")
pdf(file = paste0(output,"Ovarian_cancer_out.pdf"), width = 25, height =10)
g <- (hne + cancer_counts + mean_ratio) 
print(g)
dev.off()
```


## Pancreatic Cancer

```{r load_panceas, eval = TRUE, echo = FALSE}
vesalius <- load_visium(
    "/common/wonklab/visium_cancer/data/Pancreas/ctrl/",
    tag = "HT264P1-S1H2Fc2U1Z1Bs1-H2Bs2")

max_range <- get_max_range(vesalius)
vesalius <- vesalius %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 4, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.05, min_spatial_index = 5, verbose = F)

# force image to be 1000 pix max
vesalius <- module_score(vesalius,
    gene_list = pancreas,
    add_name = "counts")
vesalius <- compress_score(vesalius,
    trial = "counts",
    add_name = "counts_bin",
    rank = FALSE, bins = 10)

ov <- vesalius@territories$barcodes[vesalius@territories$counts > 9]
vesalius_f <- filter_assay(vesalius, cells = ov)

max_range <- get_max_range(vesalius_f)
vesalius_f <- vesalius_f %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 4, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.25, min_spatial_index = 5, verbose = F)



vesalius_f <- compute_ratio_score(vesalius_f,
    group_1 = win,
    group_2 = lose,
    score_type = "mean_ratio",
    add_name = "mean_ratio")

```




```{r pancreas_plot, eval=TRUE, echo = FALSE}
hne <- territory_plot(vesalius, trial = "Territory",
        randomise = FALSE, use_image = TRUE, alpha = 0,cex_pt = 1, cex = 15) + 
        labs(title = "H & E - Pancreatic Cancer", color = "Territories")+
        theme_void()+
        theme(
            legend.position = "None",
            plot.tag = element_text(size = 30),
            plot.margin = margin(0, 0, 0, 0, "cm"),
            plot.title = element_text(size = 15),
            legend.text = element_text(size = 15),
            legend.title = element_text(size = 15))

cancer_counts <- score_plot(vesalius, trial = "counts_bin",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 2.4, cex = 20) + 
        labs(title = "Pancreatic Cancer Marker Count", color = "Cancer Marker Counts")+
        theme_void()+
        theme(
            plot.margin = margin(1, 1, 1, 1, "cm"),
            legend.position = "right",
            plot.tag = element_text(size = 30),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

pal <- colorRampPalette(rev(c("#850101","#cd8878", "#f1f1b1","#9CAAC4","#1F3B70")))
n_groups <- length(unique(vesalius_f@territories$mean_ratio))
cols <- pal(n_groups)
mean_ratio <- territory_plot(vesalius_f, trial = "mean_ratio",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 2.4, cex = 20) + 
        labs(title = "Mean Ratio",color = "Win/Lose Ratio")+
        scale_color_manual(values = cols)+
        theme_void()+
        theme(
            plot.tag = element_text(size = 30),
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

```



```{r pancreas_plot_out, eval = TRUE, echo = FALSE, fig.width=25, fig.height=10, fig.cap = "Pancreatic Cancer"}
(hne + cancer_counts + mean_ratio) +
    plot_annotation(tag_levels = "A")
pdf(file = paste0(output,"Pancreatic_cancer_out.pdf"), width = 25, height =10)
g <- (hne + cancer_counts + mean_ratio) 
print(g)
dev.off()
```

## Breast Cancer 


```{r load_breast, eval = TRUE, echo = FALSE}
vesalius <- load_visium(
    "/common/wonklab/visium_cancer/data/CytAssist_Human_breast_Cancer/",
    tag = "CytAssist_Human_breast_Cancer")

max_range <- get_max_range(vesalius)
vesalius <- vesalius %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 2, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.05, min_spatial_index = 5, verbose = F)
# force image to be 1000 pix max
vesalius <- module_score(vesalius,
    gene_list = breast,
    add_name = "counts")
vesalius <- compress_score(vesalius,
    trial = "counts",
    add_name = "counts_bin",
    rank = FALSE, bins = 10)

ov <- vesalius@territories$barcodes[vesalius@territories$counts > 15]
vesalius_f <- filter_assay(vesalius, cells = ov)

max_range <- get_max_range(vesalius_f)
vesalius_f <- vesalius_f %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 4, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.25, min_spatial_index = 5, verbose = F)



vesalius_f <- compute_ratio_score(vesalius_f,
    group_1 = win,
    group_2 = lose,
    score_type = "mean_ratio",
    add_name = "mean_ratio")

```




```{r breast_plot, eval=TRUE, echo = FALSE}
hne <- territory_plot(vesalius, trial = "Territory",
        randomise = FALSE, use_image = TRUE, alpha = 0,cex_pt = 1, cex = 15) + 
        labs(title = "H & E - Breast Cancer", color = "Territories")+
        theme_void()+
        theme(
            legend.position = "None",
            plot.tag = element_text(size = 30),
            plot.margin = margin(0, 0, 0, 0, "cm"),
            plot.title = element_text(size = 15),
            legend.text = element_text(size = 15),
            legend.title = element_text(size = 15))

cancer_counts <- score_plot(vesalius, trial = "counts_bin",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 2.4, cex = 20) + 
        labs(title = "Breast Cancer Marker Count", color = "Cancer Marker Counts")+
        theme_void()+
        theme(
            legend.position = "right",
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.tag = element_text(size = 30),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

pal <- colorRampPalette(rev(c("#850101","#cd8878", "#f1f1b1","#9CAAC4","#1F3B70")))
n_groups <- length(unique(vesalius_f@territories$mean_ratio))
cols <- pal(n_groups)
mean_ratio <- territory_plot(vesalius_f, trial = "mean_ratio",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 2.4, cex = 20) + 
        labs(title = "Mean Ratio",color = "Win/Lose Ratio")+
        scale_color_manual(values = cols)+
        theme_void()+
        theme(
            plot.tag = element_text(size = 30),
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

```



```{r breast_plot_out, eval = TRUE, echo = FALSE, fig.width=25, fig.height=10, fig.cap ="Breast Cancer"}
(hne + cancer_counts + mean_ratio) +
    plot_annotation(tag_levels = "A")
pdf(file = paste0(output,"Breast_cancer_out.pdf"), width = 25, height =10)
g <- (hne + cancer_counts + mean_ratio)
print(g)
dev.off()
```





## Breast Cancer - Xenium


```{r load_breast_x, eval = TRUE, echo = FALSE}
vesalius <- load_visium(
    "/common/wonklab/visium_cancer/data/xenium_breast/",
    tag = "xenium_breast")


max_range <- get_max_range(vesalius)
vesalius <- vesalius %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 2, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.05, min_spatial_index = 5, verbose = F)
# force image to be 1000 pix max
vesalius <- module_score(vesalius,
    gene_list = breast,
    add_name = "counts")
vesalius <- compress_score(vesalius,
    trial = "counts",
    add_name = "counts_bin",
    rank = FALSE, bins = 10)

ov <- vesalius@territories$barcodes[vesalius@territories$counts > 19]
vesalius_f <- filter_assay(vesalius, cells = ov)

max_range <- get_max_range(vesalius_f)
vesalius_f <- vesalius_f %>%
    generate_embeddings(tensor_resolution = 1000 / max_range,
        filter_grid = 1,
        filter_threshold = 1,
        dim_reduction = "PCA",
        features = c(win,lose),
        dimensions = 10,
        verbose = FALSE) %>%
    equalize_image(dimensions = 1:7, sleft = 2, sright = 2, verbose = F) %>%
    smooth_image(dimensions = 1:7, sigma = 2, iter = 10, verbose = F) %>%
    segment_image(dimensions = 1:7, col_resolution = 5, verbose = F) %>%
    isolate_territories(capture_radius = 0.25, min_spatial_index = 5, verbose = F)



vesalius_f <- compute_ratio_score(vesalius_f,
    group_1 = win,
    group_2 = lose,
    score_type = "mean_ratio",
    add_name = "mean_ratio")

```




```{r breast_x_plot, eval=TRUE, echo = FALSE}
hne <- territory_plot(vesalius, trial = "Territory",
        randomise = FALSE, use_image = TRUE, alpha = 0,cex_pt = 1, cex = 15) + 
        labs(title = "H & E - Breast Cancer", color = "Territories")+
        theme_void()+
        theme(
            legend.position = "None",
            plot.tag = element_text(size = 30),
            plot.margin = margin(0, 0, 0, 0, "cm"),
            plot.title = element_text(size = 15),
            legend.text = element_text(size = 15),
            legend.title = element_text(size = 15))

cancer_counts <- score_plot(vesalius, trial = "counts_bin",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 1.4, cex = 20) + 
        labs(title = "Breast Cancer Marker Count", color = "Cancer Marker Counts")+
        theme_void()+
        theme(
            legend.position = "right",
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.tag = element_text(size = 30),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

pal <- colorRampPalette(rev(c("#850101","#cd8878", "#f1f1b1","#9CAAC4","#1F3B70")))
n_groups <- length(unique(vesalius_f@territories$mean_ratio))
cols <- pal(n_groups)
mean_ratio <- territory_plot(vesalius_f, trial = "mean_ratio",
        randomise = FALSE, use_image = TRUE, alpha = 1,cex_pt = 1.4, cex = 20) + 
        labs(title = "Mean Ratio",color = "Win/Lose Ratio")+
        scale_color_manual(values = cols)+
        theme_void()+
        theme(
            plot.tag = element_text(size = 30),
            plot.margin = margin(1, 1, 1, 1, "cm"),
            plot.title = element_text(size = 20),
            legend.text = element_text(size = 20),
            legend.title = element_text(size = 20))

```



```{r breast_x_plot_out, eval = TRUE, echo = FALSE, fig.width=25, fig.height=10, fig.cap ="Breast Cancer - Xenium"}
(hne + cancer_counts + mean_ratio) +
    plot_annotation(tag_levels = "A")
pdf(file = paste0(output,"Breast_2_cancer_out.pdf"), width = 25, height =10)
g <- (hne + cancer_counts + mean_ratio) 
print(g)
dev.off()
```




